src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -180,7 +180,7 @@ private unsafe Span<byte> GetReceiptData(long blockNumber, Hash256 blockHash)
             if (_legacyHashKey)
             {
                 Span<byte> receiptsData = _blocksDb.GetSpan(blockHash);
-                if (receiptsData != null)
+                if (receiptsData.Length > 0)
```

Comment:
maybe this needs a better interface, there can be 0 receipts in the block

Reply:
Can grab the underlining ponter then